src/precompile.jl

```diff
@@ -1,35 +1,29 @@
-function _precompile_()
-    ccall(:jl_generating_output, Cint, ()) == 1 || return nothing
-    # These make a difference for Makie's TTFP and should help others too
-    # Total time for this to run is ~2s. To time it, disable:
-    #   - the `_precompile_()` call in StaticArrays.jl
-    #   - the jl_generating_output line above
-    # and then do
-    #     @time StaticArrays._precompile_()
-
-    for T in (Float32, Float64, Int)   # some common eltypes
-        for S = 2:4                    # some common sizes
-            L = S*S
-            @assert precompile(Tuple{typeof(*),SMatrix{S, S, T, L},SMatrix{S, S, T, L}})
-            @assert precompile(Tuple{typeof(*),SMatrix{S, S, T, L},SVector{S, T}})
-            @assert precompile(Tuple{typeof(inv),SMatrix{S, S, T, L}})
-            @assert precompile(Tuple{typeof(transpose),SMatrix{S, S, T, L}})
-            @assert precompile(Tuple{typeof(_adjoint),Size{(S, S)},SMatrix{S, S, T, L}})
-            @assert precompile(Tuple{Core.kwftype(typeof(minimum)),NamedTuple{(:dims,), Tuple{Int}},typeof(minimum),SMatrix{S, S, T, L}})
-            @assert precompile(Tuple{Core.kwftype(typeof(maximum)),NamedTuple{(:dims,), Tuple{Int}},typeof(maximum),SMatrix{S, S, T, L}})
-            @assert precompile(Tuple{typeof(getindex),SMatrix{S, S, T, L},SOneTo{S-1},SOneTo{S-1}})
+@setup_workload begin
+    @compile_workload begin
+        for T in (Float32, Float64, Int)   # some common eltypes
+            for S = 2:4                    # some common sizes
+                L = S*S
+                x = fill(0., SMatrix{S, S, T, L})
+                x * x
+                y = fill(0., SVector{S, T})
+                x * y
+                inv(x)
+                transpose(x)
+                _adjoint(Size(S, S), x)
+                minimum(x; dims=1)
+                minimum(x; dims=2)
+                maximum(x; dims=1)
+                maximum(x; dims=2)
+                getindex(x, SOneTo(S-1), SOneTo(S-1))
+                y .* x .* y'
+                zeros(y)
+                zeros(x)
```

Comment:
@timholy see error above. I guessed

```suggestion
                zeros(T, y)
                zeros(T, x)
```
but hit 
```
ERROR: LoadError: MethodError: no method matching zeros(::Type{Float32}, ::StaticArraysCore.SVector{2, Float32})

Closest candidates are:
  zeros(::Type{T}, ::Union{Integer, AbstractUnitRange}...) where T
   @ Base array.jl:580
  zeros(::Type{T}, ::Tuple{}) where T
   @ Base array.jl:588
  zeros(::Type{T}, ::Tuple{Vararg{Integer, N}}) where {T, N}
   @ Base array.jl:583
  ...
```

Reply:
Shoot, should be `zero` not `zeros`. Sorry 'bout that!